pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1272,8 +1272,17 @@ self: super: {
   # upstream issue: https://github.com/vmchale/atspkg/issues/12
   language-ats = dontCheck super.language-ats;
 
-  # polysemy has occasional test failures from what looks like buggy async tests.
-  # We think this will probably be fixed when updating to the polysemy version in LTS-15.
-  polysemy = dontCheck super.polysemy;
+  # polysemy-plugin requires polysemy >= 1.2.0.0
+  polysemy = self.polysemy_1_2_2_0;
+
+  # The polysemy-plugin tests failed because it couldn't find
+  # the polysemy-plugin package in the doctests:
+  # https://github.com/NixOS/nixpkgs/issues/71164
+  # I've addressed this with a PR upstream:
+  # https://github.com/polysemy-research/polysemy/pull/265
+  # The patch of which is applied here.
+  polysemy-plugin = appendPatch super.polysemy-plugin ./patches/polysemy-plugin.patch;
```

Comment:
If possible, we don't like to carry around packages in the nixpkgs repo.

Also, it is generally easier for us to just disable the tests with `dontCheck` instead of trying to patch them.

Please change this to the `dontCheck` approach (or you could wait until https://github.com/polysemy-research/polysemy/pull/265 is merged and a new release is made).

Reply:
> It is generally easier for us to just disable the tests with dontCheck instead of trying to patch them.

That is true, we hardly ever patch test suites simply because that tends to be more effort than just disabling them. There is one argument to favor of patching over disabling, though: When a new version of the package comes out that incorporates the fixes, our patch will fail to apply -- which means that someone notices that the corresponding override can be deleted. That is not true for `dontCheck`; those overrides won't fail when a new version comes out and therefore we usually don't remove them even though the override has become obsolete.